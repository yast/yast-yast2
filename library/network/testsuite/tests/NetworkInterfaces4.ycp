/****************************************************************************

Copyright (c) 2002 - 2012 Novell, Inc.
All Rights Reserved.

This program is free software; you can redistribute it and/or
modify it under the terms of version 2 of the GNU General Public License as
published by the Free Software Foundation.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program; if not, contact Novell, Inc.

To contact Novell about this file by physical or electronic mail,
you may find current contact information at www.novell.com

****************************************************************************/
{
// bug 72164

include "testsuite.ycp";

map READ = $[
    "network" : $[
	"section" : $[
	    "eth4"	: nil,
	],
	"value" : $[
	    "eth4"	: $["BOOTPROTO":"dhcp", "NAME": "we like 'singles'"],
	]
    ],
    "probe" : $[
	"system" : [],
    ],
    "target" : $[
	"tmpdir" : "/tmp",
    ],
];

map EXEC = $[
    "target" : $[
        "bash_output": $[
                "exit": 0,
                "stdout": "",
                "stderr":"",
        ],
    ],
];

TESTSUITE_INIT([READ, $[], EXEC], nil);
import "NetworkInterfaces";

DUMP("NetworkInterfaces::Read");
TEST(``(NetworkInterfaces::Read()), [READ, $[], EXEC], nil);
NetworkInterfaces::OriginalDevices = nil;

DUMP("NetworkInterfaces::Write");
TEST(``(NetworkInterfaces::Write("")), [READ], nil);
}
