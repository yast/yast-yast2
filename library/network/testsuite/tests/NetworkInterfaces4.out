Read	.target.tmpdir "/tmp"
Dump	NetworkInterfaces::Read
Dir	.network.section: ["eth4"]
Dir	.network.value."eth4": ["BOOTPROTO", "NAME"]
Read	.network.value."eth4".BOOTPROTO "dhcp"
Read	.network.value."eth4".NAME "we like 'singles'"
Read	.target.stat "/sys/class/net/eth4/type" nil
Read	.target.string "/sys/class/net/eth4/type" nil
Return	true
Dump	NetworkInterfaces::Write
Write	.network.value."eth4".BOOTPROTO "dhcp" true
Write	.network.value."eth4".NAME "we like 'singles'" true
Write	.network.value."eth4".STARTMODE "" true
Write	.network nil true
Return	true
